slack_sdk/web/async_client.py

```diff
@@ -2992,6 +2995,160 @@ async def files_upload(
+        content: Optional[Union[str, bytes]] = None,
+        title: Optional[str] = None,
+        alt_txt: Optional[str] = None,
+        snippet_type: Optional[str] = None,
+        filetype: Optional[str] = None,  # no longer supported
+        # To upload multiple files at a time
+        upload_files: Optional[List[Dict[str, Any]]] = None,
+        channel: Optional[str] = None,
+        channels: Optional[Union[str, Sequence[str]]] = None,  # having n channels is no longer supported
+        initial_comment: Optional[str] = None,
+        thread_ts: Optional[str] = None,
+        **kwargs,
+    ) -> AsyncSlackResponse:
+        """This wrapper method provides an easy way to upload files using the following endpoints:
+        * step1: https://api.slack.com/methods/files.getUploadURLExternal
+        * step2: "https://files.slack.com/upload/v1/..." URLs returned from files.getUploadURLExternal API
+        * step3: https://api.slack.com/methods/files.completeUploadExternal
+        """
+        if file is None and content is None and upload_files is None:
+            raise e.SlackRequestError("Any of file, content, and upload_files must be specified.")
+        if file is not None and content is not None:
+            raise e.SlackRequestError("You cannot specify both the file and the content argument.")
+        if channels is not None and (
+            (isinstance(channels, (list, Tuple)) and len(channels) > 1)
+            or (isinstance(channels, str) and len(channels.split(",")) > 1)
+        ):
+            raise e.SlackRequestError("Sharing files with multiple channels is no longer supported")
+        if filetype is not None:
+            warnings.warn("filetype is no longer supported. Please remove it from the arguments.")
+
+        # step1: files.getUploadURLExternal per file
+        files: List[Dict[str, Any]] = []
+        if upload_files is not None:
+            for f in upload_files:
+                files.append(_to_v2_upload_file_item(f))
+        else:
+            f = _to_v2_upload_file_item(
+                {
+                    "filename": filename,
+                    "file": file,
+                    "content": content,
+                    "title": title,
+                    "alt_txt": alt_txt,
+                    "snippet_type": snippet_type,
+                }
+            )
+            files.append(f)
+
+        for f in files:
+            url_response = await self.files_getUploadURLExternal(
+                filename=f.get("filename"),
+                length=f.get("length"),
+                alt_txt=f.get("alt_txt"),
+                snippet_type=f.get("snippet_type"),
+                token=kwargs.get("token"),
+            )
+            f["file_id"] = url_response.get("file_id")
+            f["upload_url"] = url_response.get("upload_url")
+
+        # step2: "https://files.slack.com/upload/v1/..." per file
+        for f in files:
+            upload_result = _upload_file_via_v2_url(
```

Comment:
We don't use aiohttp for the requests uploading files so far and I believe that this is okay. But if we receive many requests to use only aiohttp for some reason, we may revisit this.